google_guest_agent/network/manager/common.go

```diff
@@ -116,17 +117,19 @@ func interfaceListsIpv4Ipv6(nics []metadata.NetworkInterfaces) ([]string, []stri
 
 	for _, ni := range nics {
 		iface, err := GetInterfaceByMAC(ni.Mac)
+		ifaceName := iface.Name
 		if err != nil {
 			if _, found := badMAC[ni.Mac]; !found {
 				logger.Errorf("error getting interface: %s", err)
 				badMAC[ni.Mac] = iface
 			}
-			continue
+			// Mark the iface as invalid, and include its Mac.
+			ifaceName = fmt.Sprintf("invalid-%s", ni.Mac)
```

Comment:
describe why it's important to include this interface in the slice even though it's a invalid one.